src/components/common/Sign/SignForm.tsx

```diff
@@ -0,0 +1,110 @@
+import { useState } from 'react'
+import { Link, useNavigate } from 'react-router-dom'
+
+import { StyledSignFormContainer, StyledSignFormTitle, StyledSignFormBox } from './SignForm.styled'
+import { postSignUp, postSignin } from '../../../apis/auth'
+
+type Props = {
+  isSignUp: boolean
+  setIsAuth: React.Dispatch<React.SetStateAction<string>>
+}
+
+function SignForm({ isSignUp, setIsAuth }: Props) {
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const [isValidEmail, setIsValidEmail] = useState(false)
+  const [isValidPassword, setIsValidPassword] = useState(false)
+  const navigate = useNavigate()
+
+  const onChangeEmail = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value
+    setEmail(value)
+    setIsValidEmail(/@/.test(value))
+  }
+
+  const onChangePassword = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value
+    setPassword(value)
+    setIsValidPassword(/.{8,}$/.test(value))
+  }
+
+  const sendData = () => {
+    const data = {
+      email: email,
+      password: password,
+    }
+    if (isSignUp) {
+      postSignUp(data.email, data.password)
+        .then(() => {
+          alert('회원가입 완료!')
+          navigate('/signin')
+        })
+        .catch((err) => {
+          const message = err.response.data.message
+          alert(message)
+          console.log(err)
+        })
+    }
+
+    if (isSignUp === false) {
+      postSignin(data.email, data.password)
+        .then((res) => {
+          localStorage.setItem('access_token', res.data.access_token)
+          setIsAuth(res.data.access_token)
+          navigate('/todo')
+        })
+        .catch((err) => {
+          const message = err.response.data.message
+          alert(message)
+          console.log(err)
+        })
+    }
+  }
+
+  return (
+    <StyledSignFormContainer>
+      <StyledSignFormTitle>{isSignUp ? '회원가입' : '로그인'}</StyledSignFormTitle>
+      <StyledSignFormBox>
```

Comment:
```suggestion
      <StyledSignFormBox noValidate>
```

form 태그를 사용하시면 기본적으로 유효성검사가 됩니다. noValidate 속성을 사용해서 꺼주시면 더 좋을 것 같아요.

![image](https://github.com/wanted-pre-onboarding-team-12th-7/pre-onboarding-12th-1-7/assets/97281800/c715786d-a37b-4e1b-9da1-9e6bf351b519)